addons/website/models/website.py

```diff
@@ -600,7 +600,22 @@ def _image(self, cr, uid, model, id, field, response, max_width=maxint, max_heig
         if response.status_code == 304:
             return response
 
-        data = record[field].decode('base64')
+        if model == 'ir.attachment' and field == 'url':
+            path = record.get(field).strip('/')
+
+            # Check that we serve a file from within the module
+            if os.path.normpath(path).startswith('..'):
+                return self._image_placeholder(response)
+
+            # Check that the file actually exists
+            path = path.split('/')
+            resource = openerp.modules.get_module_resource(path[0], *path[1:])
+            if not resource:
+                return self._image_placeholder(response)
+
+            data = open(resource, 'rb').read()
+        else:
+            data = record[field].decode('base64')
         image = Image.open(cStringIO.StringIO(data))
```

Comment:
Move the initialisation of the image object to the conditionals, no need to take data out of a file to put it into a StringIO so it can be used by Pillow.

Plus Image.open can take the filename directly, manually opening the file isn't even necessary.